### PR TITLE
Adds 420's new plant mutations

### PR DESCRIPTION
**Pull request by TZK13**
[Read about all of the plant mutations being added here in this thread.](https://tgstation13.org/phpBB/viewtopic.php?f=11&t=2435)

The largest difference from what's outlined above is with what I've named the panana (It's a pun on pantomime also upside down b), a 99.9% invisible banana. Now in order to curtail the immediate amount of FUN that's conjured up in your mind from that the also invisible panana peel is only good for one use before it disappears as if it was never there to begin with. Pananas are also a good source of nothing and mute toxin, note that in order to get a panana peel you must first eat the panana. This is also a rather ingenious way to add a new plant without adding anything at all to growing.dmi.

OTHER MINOR FEATURES:
-You can now mutate cabbage into replica pods as another source of replica pod seeds in case you run out.
-Novaflowers now ignite people with the amount of fire stacks applied depending on potency.

I've also cleaned and split the hydroponics.dmi into growing.dmi and equipment.dmi as I ran into the limit of icons while working on this PR. All dmi files related to hydroponics have been placed inside of a new hydroponics folder as suggested by Cheridan.

### CODE REVIEW

**Comment by Ergovisavi** [comment]
>An instant AoE stun that nothing in the game except magboots can save you from

Please no

Other than that, seems fine.

**Comment by Dorsisdwarf** [comment]
>45 burn damage weapons that disintegrate and leave no trace.
>Enough chems to turn you into ghetto changeling.
>Lube tomatoes.
>Gib tomatos which are literally throwable griff.

File: code/modules/hydroponics/grown.dm [hunk 1]
```diff
@@ -499,8 +499,8 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/grown/icepepper
 	seed = /obj/item/seeds/icepepperseed
-	name = "ice-pepper"
-	desc = "It's a mutant strain of chili"
+	name = "chilly pepper"
+	desc = "It's cold to the touch let alone to your taste buds."
```
Comment (Dorsisdwarf) [inline code/modules/hydroponics/grown.dm]: Should be "It's cold to the touch, let alone to your taste buds"

File: code/modules/hydroponics/grown.dm [hunk 2]
```diff
@@ -518,7 +518,7 @@
 
 /obj/item/weapon/reagent_containers/food/snacks/grown/ghost_chilli
 	seed = /obj/item/seeds/chillighost
-	name = "ghost chili"
+	name = "ghost pepper"
```
Comment (Firecage) [inline code/modules/hydroponics/grown.dm]: Huh, wait, why did you change this to ghost pepper? It is a chili.

File: code/modules/hydroponics/grown.dm (hunk 1 above)
Comment (Miauw62) [inline code/modules/hydroponics/grown.dm]: this rename is great.

**Comment by Chocobro** [comment]
what happened to "im gonna add all your plants and more" tzk
what happened?????

File: code/modules/hydroponics/grown.dm (hunk 1 above)
Comment (dannno) [inline code/modules/hydroponics/grown.dm]: how did no one think of this already

File: code/modules/hydroponics/grown.dm (hunk 2 above)
Comment (TZK13) [inline code/modules/hydroponics/grown.dm]: A chili shouldn't be called a chili, they're chili peppers. Same logic with the bhut jolokia, if you look at the search result numbers for "ghost pepper", "ghost chili pepper", and "ghost chili" you'll see that ghost pepper has the highest number.

**Comment by TZK13** [comment]
I wanted to at least get this PR up already because I had been attempting it a few times. I'm also still concerned about the icon limit in the new growing.dmi because there's still quite a few in there.

File: code/modules/hydroponics/grown.dm (hunk 1 above)
Comment (TZK13) [inline code/modules/hydroponics/grown.dm]: I just realized the description is stupid because your taste buds have nothing to do with "tasting" how cold food is :^)

Gonna change it to more closely resemble the chili pepper description.

**Comment by TZK13** [comment]
I actually agree with Ergo about how OP a practically unavoidable AoE stun that can be rather easily mass produced and is easy to chain together thus I've removed the stunning effect from it.

**Comment by Chocobro** [comment]
:-1: on pananas / what is wrong with mimanas / why aren't you adding the rest of my shit

**Comment by TZK13** [comment]
Mimanas sounds silly, panana is the better pun.

>pantomime
 noun
1.
the art or technique of conveying emotions, actions, feelings, etc., by gestures without speech.

**Comment by TZK13** [comment]
I'll go through the others and why I haven't or aren't going to add them:

Borschtbeet/Oat/Sweet Potato/Vanilla: Fucking wowie real world boring ass plants that are hardly "mutations" for recipes we either already have or need sprites for.

Geranium/Lily: Recoloured poppies, literally what is the point. If we're going to add more flowers they should not just be recolours.

Blue Cherries: You said it yourself, they currently have no purpose in mind and we have enough blue shit already.

Parsnip: It's a fairly vague and dumb use honestly.

Snapcorn: I like the idea enough but I don't really like the sprites, they're barely discernible from normal corn and if it's growing snap pops why isn't it grey or something?

Shantyrelle: I don't know how to really go about adding that sort of functionality.

Rice/Steelcaps: As I said before I do intend to add these.

**Comment by Iamgoofball** [comment]
shantyrelle can be a reagent effect

**Comment by TZK13** [comment]
That's what I viewed as most sensible, I still don't know how to go about it though.

**Comment by TZK13** [comment]
A personal disagreement between me and 420weedscopes has resulted in him requesting that I remove the plumpkin and holymelon which I have thus done.

**Comment by Ergovisavi** [comment]
I have a slight concern in that having an invisible banana and a chameleon projector results in a completely invisible, 100% undetectable traitor.

File: code/modules/hydroponics/grown.dm [hunk 3]
```diff
@@ -1119,15 +1118,71 @@ obj/item/weapon/reagent_containers/food/snacks/grown/shell/eggy/add_juice()
...
+				var/mob/living/carbon/human/H = X
+				if(istype(H.shoes,/obj/item/clothing/shoes/magboots))
+					var/obj/item/clothing/shoes/magboots/M = H.shoes
+					if(M.magpulse)
+						continue
+				H.apply_effect(min(3, vortexpower))
+				step_towards(H,pull)
+				step_towards(H,pull)
+				step_towards(H,pull)
+				step_towards(H,pull)
```
Comment (Ergovisavi) [inline code/modules/hydroponics/grown.dm]: Can you change these multiple step_towards to a for loop? Just makes it look nicer.

File: code/modules/hydroponics/grown.dm [hunk 4]
```diff
@@ -1119,15 +1118,71 @@ obj/item/weapon/reagent_containers/food/snacks/grown/shell/eggy/add_juice()
...
+				step_towards(X,pull)
+				step_towards(X,pull)
+				step_towards(X,pull)
+			else if(istype(X,/mob/living/carbon/human))
+				var/mob/living/carbon/human/H = X
+				if(istype(H.shoes,/obj/item/clothing/shoes/magboots))
+					var/obj/item/clothing/shoes/magboots/M = H.shoes
+					if(M.magpulse)
+						continue
+				H.apply_effect(min(3, vortexpower))
```
Comment (Ergovisavi) [inline code/modules/hydroponics/grown.dm]: This line has no affect, apply_effect requires a string in its first var to do anything

**Comment by Jordie0608** [comment]
Changelog

**Comment by TZK13** [comment]
Restructuring this PR again after coming to an agreement with 420weedscopes, this PR now concentrates on his plants. I'll address singulo potatoes again in the future maybe.

**Comment by TZK13** [comment]
I would be most interested in what @Cheridan has to say in regards to these plant mutations considering it was him that prompted 420weedscopes to make them.

**Comment by TZK13** [comment]
In regards to a chameleon projector being paired with an invisible banana peel I can make them less completely transparent so that while you still probably won't see one sitting in your path you'd be more likely to see it moving around.

**Comment by Ergovisavi** [comment]
It might be worth simply changing their invisibility to SEE_INVISIBLE_MINIMUM, this way they are  detected by people wearing mesons, chameleon projectors won't inherit the invisibility, and you still have an invisible banana.

**Comment by TZK13** [comment]
Sounds good, will do later.

**Comment by Chocobro** [comment]
tzk I don't know if you checked but there is an updated dmi that actually contains the lily inhand, link is in my thread
glhf

**Comment by Chocobro** [comment]
also another thing I just remembered: invisible sprites cannot be clicked on, you have to analyze (right click) the whole tile to see it - in other words, make sure there is at least something to click for mimanas.

**Comment by Chocobro** [comment]
er, pananas*
(sorry for new post, phone git cannot edit comments)

**Comment by TZK13** [comment]
The sprites for it weren't actually invisible, they just had a low alpha level so you could still see them if you put your mouse over them. I'll see about all this when I update the PR which I'll probably do tomorrow.

**Comment by Chocobro** [comment]
ah okay - so long as they're not 100% invisible it should be good to go. maybe 5-10% alpha.

**Comment by Chocobro** [comment]
still waiting on you to update this @TZK13

**Comment by TZK13** [comment]
Considering how this PR has changed quite considerably, has fallen rather far back, and is a mess at this point I'm going to just close it and make a new PR with everything here currently and the things it needs to be updated with.
